lib/synapse/flat_file.rb

```diff
@@ -0,0 +1,53 @@
+require 'synapse/log'
+require 'fileutils'
+require 'tempfile'
+
+module Synapse
+  class FlatFile
+    include Logging
+    attr_reader :opts, :name
+
+    def initialize(opts)
+      super()
+
+      unless opts.has_key?("output_directory")
+        raise ArgumentError, "flat file generation requires an output_directory key"
+      end
+
+      begin
+        FileUtils.mkdir_p(opts['output_directory'])
```

Comment:
What's the rationale for creating the output directory if it doesn't already exist?  Is it better to delegate that behavior to Puppet (say)?